scripts/test/terra_validation.py

```diff
@@ -0,0 +1,135 @@
+#!/bin/python
+
+import subprocess
+import argparse
+import json
+from os import listdir
+from os.path import basename, isfile
+import logging
+import atexit
+
+
+"""
+Summary: Cursory validation of Terra input JSONs for pipeline workflows and metrics workflows. 
+  Checks if JSONs contain all required inputs and that they do not contain extraneous inputs.
+  Does not perform type-checking. Also checks that all expected JSONs have been generated.
+
+Usage:
+  python scripts/test/terra_validation.py -d /path/to/base/dir -j /path/to/womtool/jar [optional flags]
+
+Parameters:
+  /path/to/base/dir: path to base directory of gatk-sv repo
+  /path/to/womtool/jar: path to user's womtool jar file
+  Optional flags:
+  --log-level LEVEL (specify level of logging information to print, ie. INFO, WARNING, ERROR - not case-sensitive)
+
+Outputs: If successful, last line of printout should read "19 of 19 Terra input JSONs exist and passed validation."
+  Prior lines will detail the JSONs that were examined and any errors found.
+"""
+
+
+TMP_JSON = "womtool_inputs.tmp.json"
+WDLS_PATH = "/wdl/"
+TERRA_INPUTS_PATH = "/inputs/terra_workspaces/cohort_mode/workflow_configurations/"
+METRICS_INPUTS_PATH = "metrics_workflows/"
+NUM_TERRA_INPUT_JSONS = 12
+NUM_TERRA_METRICS_JSONS = 7
+
+
+def exit_handler():
+  if isfile(TMP_JSON):
+    subprocess.run("rm " + TMP_JSON, shell=True)
+
+
+def list_jsons(inputs_path):
+  jsons = [x for x in listdir(inputs_path) if x.endswith(".json")]
+  num_input_jsons = len(jsons)
+  if num_input_jsons < NUM_TERRA_INPUT_JSONS:
+    logging.warning(f"Expected {NUM_TERRA_INPUT_JSONS} Terra input JSONs but found {num_input_jsons}.")
+  jsons.sort()
+
+  metrics_jsons = [METRICS_INPUTS_PATH + x for x in listdir(inputs_path + METRICS_INPUTS_PATH) if x.endswith(".json")]
+  num_metrics_jsons = len(metrics_jsons)
+  if num_metrics_jsons < NUM_TERRA_METRICS_JSONS:
+    logging.warning(f"Expected {NUM_TERRA_METRICS_JSONS} Terra metrics input JSONs but found {num_metrics_jsons}.")
+  metrics_jsons.sort()
```

Comment:
This looks a lot like the block above. Can you create a common function to replace both blocks?